supervisor/process.py

```diff
@@ -123,6 +123,8 @@ def get_execv_args(self):
                 st = None
 
         else:
+            if self.config.environment is not None:
+                os.environ.update(self.config.environment)
```

Comment:
The function you are changing, `get_execv_args()`, [runs in the `supervisord` process](https://github.com/Supervisor/supervisor/blob/154cb4c84f28ac3e0ac1ce5409faea65c15d2d02/supervisor/rpcinterface.py#L279).  If this change was made, the `environment=` of any `[program:x]` section would leak into `supervisord` and then into all subprocesses.  The expected behavior is that the `environment=` of a `[program:x]` affects only that program.